src/Microsoft.AspNet.Mvc.Core/ActionConstraints/IConsumesActionConstraint.cs

```diff
@@ -0,0 +1,17 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.AspNet.Mvc.HeaderValueAbstractions;
+
+namespace Microsoft.AspNet.Mvc
+{
+    /// <summary>
+    /// An <see cref="IActionConstraint"/> constraint that identifies a type which can be used to select an action
+    /// based on incoming request.
+    /// </summary>
+    public interface IConsumesActionConstraint : IActionConstraint
```

Comment:
I'm not 100% comfortable with the derivation here. Should we just have them side by side?